src/main/java/io/cdap/plugin/servicenow/source/ServiceNowRecordReader.java

```diff
@@ -43,16 +43,20 @@
 public class ServiceNowRecordReader extends RecordReader<NullWritable, StructuredRecord> {
   private static final Logger LOG = LoggerFactory.getLogger(ServiceNowRecordReader.class);
   private final ServiceNowSourceConfig pluginConf;
-  private ServiceNowInputSplit split;
-  private int pos;
-  private List<Schema.Field> tableFields;
-  private Schema schema;
-
-  private String tableName;
-  private String tableNameField;
-  private List<Map<String, Object>> results;
-  private Iterator<Map<String, Object>> iterator;
-  private Map<String, Object> row;
+  protected ServiceNowInputSplit split;
+  protected int pos;
+  protected List<Schema.Field> tableFields;
+  protected Schema schema;
+
+  protected String tableName;
+  protected String tableNameField;
+  protected List<Map<String, Object>> results;
+  protected Iterator<Map<String, Object>> iterator;
+  protected Map<String, Object> row;
+
+  public ServiceNowRecordReader() {
```

Comment:
why do you need this constructor? No need to initialize pluginConf to null

Reply:
This class `ServiceNowMultiRecordReader` extends `ServiceNowRecordReader` class.

The base class constructor :  
`public ServiceNowRecordReader() {
    pluginConf = null;
  }`

is invoked from the derived class constructor : 

`ServiceNowMultiRecordReader(ServiceNowMultiSourceConfig multiSourcePluginConf) {
    super();
    this.multiSourcePluginConf = multiSourcePluginConf;
  }`
  
  by calling `super()`
  
I had to implement it this way as I was getting a compilation error.

Reply:
hmm then why is ServiceNowMultiRecordReader extending ServiceNowRecordReader? Seems unnecessary if it's not really using methods implemented by ServiceNowRecordReader

Reply:
It is using this method from the base class: `ServiceNowRecordReader.convertToValue(String fieldName, Schema fieldSchema, Map<String, Object> record)` 

Also, the return type of the method : `ServiceNowMultiInputFormat.createRecordReader` happens to be an object of class type: `RecordReader<NullWritable, StructuredRecord>` so we have to return a new instance of this class type. 

So I had to either directly extend the `RecordReader` class or extend `ServiceNowRecordReader` class which was further extending `RecordReader` class. i chose the latter as it involved implementing less number of methods.

Please let me know if you feel we can write this piece of code in a better way.